dev/io.openliberty.cdi.4.0.internal_fat/fat/src/io/openliberty/cdi40/internal/fat/startupEvents/sharedLib/AbstractObserver.java

```diff
@@ -149,6 +159,28 @@ private void runAllJNDITests() throws Exception {
         testBeanJNDILookup();
     }
 
+    private void runTCCLTest() throws Exception {
+
+        ClassLoader tccl = Thread.currentThread().getContextClassLoader();
+        String id = "unset";
+
+        //This duplicates the logic from com.ibm.ws.classloading.internal.ClassLoadingServiceImpl
+        if (tccl.getClass().getName().contains("ThreadContextClassLoader")) {
+            Method getID = tccl.getClass().getMethod("getKey");
+            id = (String) getID.invoke(tccl);
+        } else if (tccl.getClass().getName().contains("ThreadContextClassLoader")) {
+            Method getID = tccl.getClass().getMethod("getKey");
+            Object classLoaderIdentity = getID.invoke(tccl);
+            id = classLoaderIdentity.toString();
+        } else {
+            Assert.fail("TCCL was neither a ThreadContextClassLoader or an AppClassLoader");
+        }
```

Comment:
1. both `if` conditions appear to check for the same thing
3. The assert message suggests that either `ThreadContextClassLoader` or `AppClassLoader` is acceptable - shouldn't it always be `ThreadContextClassLoader`?
2. I don't see what logic from ClassLoadingServiceImpl this code is duplicating
3. If we know that the classloader is a `ThreadContextClassLoader`, would it be sufficient to ensure it can load a resource from the application to check that we have the right one?

Reply:
1 & 2. My mistake. The second if should be for AppCllassLoader
3. The method `getClassLoaderIdentifier`
4. I'll ask the class loading people to make sure there's no possibility of getting a false positive that way.

Reply:
3. aha, I see it, thank you